From f1b3aa789ea973bb0a98362f7de5a8aaa4e06090 Mon Sep 17 00:00:00 2001
From: Bogdan Hamciuc <bogdan.hamciuc@freescale.com>
Date: Thu, 23 Apr 2015 18:28:48 +0300
Subject: [PATCH 197/452] fsl-dpaa2: mac: Change IRQ flags

Signed-off-by: Bogdan Hamciuc <bogdan.hamciuc@freescale.com>
Change-Id: Ia86570858f9cf7f673089cd7c2078662d56b2f01
Reviewed-on: http://git.am.freescale.net:8181/35581
Tested-by: Review Code-CDREVIEW <CDREVIEW@freescale.com>
Reviewed-by: Stuart Yoder <stuart.yoder@freescale.com>
[Xulin: Original patch taken from FSL LS2085 SDK EAR6.0,
LS2085A-SDK-SOURCE-20160304-yocto.iso]
Signed-off-by: Xulin Sun <xulin.sun@windriver.com>
---
 drivers/staging/fsl-dpaa2/mac/mac.c |    3 ++-
 1 files changed, 2 insertions(+), 1 deletions(-)

diff --git a/drivers/staging/fsl-dpaa2/mac/mac.c b/drivers/staging/fsl-dpaa2/mac/mac.c
index 559842a..9c1617b 100644
--- a/drivers/staging/fsl-dpaa2/mac/mac.c
+++ b/drivers/staging/fsl-dpaa2/mac/mac.c
@@ -481,7 +481,8 @@ static int ppx_setup_irqs(struct fsl_mc_device *mc_dev)
 
 	err = devm_request_threaded_irq(&mc_dev->dev,
 					mc_dev->irqs[0]->irq_number,
-					NULL, &ppx_irq_handler, 0,
+					NULL, &ppx_irq_handler,
+					IRQF_NO_SUSPEND | IRQF_ONESHOT,
 					dev_name(&mc_dev->dev), &mc_dev->dev);
 	if (err) {
 		dev_err(&mc_dev->dev, "devm_request_threaded_irq err %d\n",
-- 
1.7.5.4

